graylog2-web-interface/src/components/grok-patterns/GrokPatterns.jsx

```diff
@@ -22,60 +24,100 @@ const GrokPatternsList = styled(DataTable)`
   }
 `;
 
-const GrokPatterns = createReactClass({
-  displayName: 'GrokPatterns',
+class GrokPatterns extends React.Component {
+  constructor(props) {
+    super(props);
 
-  getInitialState() {
-    return {
+    this.state = {
       patterns: [],
+      pagination: {
+        page: 1,
+        perPage: 10,
+        count: 0,
+        total: 0,
+        query: '',
+      },
     };
-  },
+  }
 
   componentDidMount() {
     this.loadData();
-  },
+  }
 
   componentWillUnmount() {
     if (this.loadPromise) {
       this.loadPromise.cancel();
     }
-  },
+  }
 
-  loadData() {
-    this.loadPromise = GrokPatternsStore.loadPatterns((patterns) => {
-      if (!this.loadPromise.isCancelled()) {
-        this.loadPromise = undefined;
+  loadData = (callback) => {
+    const { pagination: pagination1 } = this.state;
+    const { page, perPage, query } = pagination1;
```

Comment:
Could this just be?
```suggestion
    const { pagination: { page, perPage, query } } = this.state;
```

Reply:
I have nothing against extracting on two lines personally. For me it's more readable than extracting inline. I know it's personal preference, but I've also heard newcomers to be confused by the double deconstruct in one line.